### DISCUSSION TITLE
How to create new variable for reused register?

### DISCUSSION THREAD

**Discussion by landaire** [Vector35/binaryninja-api#2158] [General] [answered]
I have the following HLIL where `*cx` is used as both the input and output register:

```
struct FILE* replay_file = bytecode_cache->replay_block_file
if (replay_file == 0)
    replay_file = nullptr
else if (bytecode_cache->packet_data == 0)
    replay_file = ftell(replay_file)
else
    replay_file = zx.q(bytecode_cache->bytes_read.d)
```

Each of these assignments in the conditional bodies should ideally be assigned to another pseudo-variable or a different type from this point forward. How can I create a new pseudo-variable to help make my the semantics of the HLIL more clear? Something like:

```
struct FILE* replay_file = bytecode_cache->replay_block_file
uint64_t bytes_read
if (replay_file == 0)
    bytes_read = 0
else if (bytecode_cache->packet_data == 0)
    bytes_read = ftell(replay_file)
else
    bytes_read = zx.q(bytecode_cache->bytes_read.d)
```

**Comment by psifertex** [comment, marked as answer]
So the short summary is that we agree this is a valuable capability we don't currently have, but it's a fairly large task given our current implementation. We're tracking it in #1887 but unfortunately don't have an ETA or spot on the roadmap for it at this time.
